charmander/enums.py

```diff
@@ -0,0 +1,79 @@
+"""
+Platform enums.
+
+"""
+from enum import Enum, unique
+
+
+@unique
+class Purpose(Enum):
```

Comment:
In our environment, we have found two enums to be extremely useful: Purpose and Resolution.

Purpose lets us tag resources for different business purposes, e.g. testing, demoing, etc and makes it easier for the system and humans to know what they are dealing with.

We very rarely truly delete anything, but we often change the `resolution` to reflect things going out of scope. The `Resolution` enum tracks this transition as well as the reason for the transition.